engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -4564,18 +4574,23 @@ private VMInstanceVO orchestrateReConfigureVm(final String vmUuid, final Service
                 _capacityMgr.allocateVmCapacity(vm, false); // lock the new capacity
             }
 
-            final Answer reconfigureAnswer = _agentMgr.send(vm.getHostId(), reconfigureCmd);
-            if (reconfigureAnswer == null || !reconfigureAnswer.getResult()) {
-                s_logger.error("Unable to scale vm due to " + (reconfigureAnswer == null ? "" : reconfigureAnswer.getDetails()));
-                throw new CloudRuntimeException("Unable to scale vm due to " + (reconfigureAnswer == null ? "" : reconfigureAnswer.getDetails()));
+            Answer scaleVmAnswer = _agentMgr.send(vm.getHostId(), scaleVmCommand);
+            if (scaleVmAnswer == null || !scaleVmAnswer.getResult()) {
+                String msg = String.format("Unable to scale %s due to [%s].", vm.toString(), (scaleVmAnswer == null ? "" : scaleVmAnswer.getDetails()));
+                s_logger.error(msg);
+                throw new CloudRuntimeException(msg);
             }
             if (vm.getType().equals(VirtualMachine.Type.User)) {
                 _userVmMgr.generateUsageEvent(vm, vm.isDisplayVm(), EventTypes.EVENT_VM_DYNAMIC_SCALE);
             }
             success = true;
-        } catch (final OperationTimedoutException e) {
-            throw new AgentUnavailableException("Operation timed out on reconfiguring " + vm, dstHostId);
-        } catch (final AgentUnavailableException e) {
+        } catch (OperationTimedoutException e) {
+            String msg = String.format("Unable to scale %s due to [%s].", vm.toString(), e.getMessage());
+            s_logger.error(msg, e);
+            throw new AgentUnavailableException(msg, dstHostId);
+        } catch (AgentUnavailableException e) {
+            String msg = String.format("Unable to scale %s due to [%s].", vm.toString(), e.getMessage());
+            s_logger.error(msg, e);
             throw e;
         } finally {
```

Comment:
log, handle or rethrow, pick one i'd say @GutoVeronezi Why do you print the stack and rethrow?

Reply:
@DaanHoogland lack of attention, perhaps. Done.